### ISSUE TITLE
Accessing full method name

### ISSUE THREAD

**Issue by lanttu** [malijs/mali#5]
I need to access the fully fledged method name `/[package].[service]/[method]` in a middleware. That can be done either by adding a new property to `Context` or repurposing `name` property to contain the full name. What do you think, which one is the better approach?

**Comment by bojand** [comment]
Hello,

Thanks for the suggestion, and I agree this would be useful information to have available within the `context`. I am thinking maybe `path` might be a more appropriate property name to use for this? And that way `name` can stay as is and `path` would be something like `helloworld.Greeter/sayHello` or whatever.

Bojan

**Comment by bojand** [comment]
I can probably get to this later this week.

**Comment by lanttu** [comment]
I'm Okay with `path`. Looks like it's been used in [gRPC over HTTP2 specs](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#appendix-a---grpc-for-protobuf). For some reason [grpc-go](https://godoc.org/google.golang.org/grpc#StreamServerInfo) uses `FullMethod` and so does [grpc-java](http://www.grpc.io/grpc-java/javadoc/io/grpc/MethodDescriptor.html#getFullMethodName--).

**Comment by bojand** [comment]
OK I see. Consistency with the other platforms in this case seems appropriate, so I think `fullMethod` or `fullMethodName` might be better.

**Comment by ojkelly** [comment]
I've found `name` useful, but after some thought I think it could be useful to access them all. Which is to say being able to get the proto package name, the service name and the method name.

From the link above:
> The service interfaces declared by protobuf are easily mapped onto GRPC by code generation extensions to protoc. The following defines the mapping to be used
>
>Path → / Service-Name / {method name}
>Service-Name → ?( {proto package name} "." ) {service name}
>Message-Type → {fully qualified proto message name}
>Content-Type → "application/grpc+proto"

**Comment by bojand** [comment]
Started thinking / working on this. Would this be good for context?

```
{ 
  type: 'unary', // unchanged, one of Mali Call types
  name: 'SayHello', // unchanged, the name of the call
  fullName: '/helloworld.Greeter/SayHello', // NEW - full name
  serviceName: 'Greeter', // NEW - service name
  packageName: 'helloworld', // NEW - package name
  requestName: 'HelloRequest', // NEW - name of the request message
  responseName: 'HelloReply', // NEW - name of the response message
  metadata: ..., // unchanged metadata object
  req: ..., // unchanged req object
  res: ..., // unchanged, res object if applicable, only in case of Duplex call
  call: ..., // unchanged, the internal call object
  app: ... // unchanged the app reference
}
```

Please provide feedback. Should have something in a few days.

**Comment by ojkelly** [comment]
@bojand I think that covers it pretty well.

**Comment by bojand** [comment]
Merged and published as 0.2.3.
